cli/bpmetadata/tfconfig.go

```diff
@@ -0,0 +1,257 @@
+		if rootBlock.Type == "terraform" {
+			tfContent, _, tfContentDiags := rootBlock.Body.PartialContent(metaSchema)
+			diags = append(diags, tfContentDiags...)
+			for _, tfContentBlock := range tfContent.Blocks {
+				if tfContentBlock.Type == "provider_meta" {
+					metaContent, _, metaContentDiags := tfContentBlock.Body.PartialContent(metaBlockSchema)
+					diags = append(diags, metaContentDiags...)
+					if versionAttr, defined := metaContent.Attributes["module_name"]; defined {
+						var modName string
+						gohcl.DecodeExpression(versionAttr.Expr, nil, &modName)
+						re := regexp.MustCompile(versionRegEx)
+						m := re.FindStringSubmatch(modName)
+						if len(m) > 0 {
+							return m[len(m)-1], diags
+						}
+
+						return "", diags
+					}
+				}
+			}
+			break
+		}
+	}
+
+	return "", diags
+}
+
+// getBlueprintInterfaces gets the variables and outputs associated
+// with the blueprint
+func getBlueprintInterfaces(configPath string) *BlueprintInterface {
+	//load the configs from the dir path
+	mod, diags := tfconfig.LoadModule(configPath)
+	for _, diag := range diags {
+		if diag.Severity == tfconfig.DiagError {
+			return nil
+		}
+	}
+
+	var variables []BlueprintVariable
+	for _, val := range mod.Variables {
+		v := getBlueprintVariable(val)
+		variables = append(variables, v)
+	}
+
+	var outputs []BlueprintOutput
+	for _, val := range mod.Outputs {
+		o := getBlueprintOutput(val)
+
+		outputs = append(outputs, o)
+	}
+
+	i := &BlueprintInterface{
+		Variables: variables,
+		Outputs:   outputs,
+	}
+
+	return i
+}
+
+//build variable
+func getBlueprintVariable(modVar *tfconfig.Variable) BlueprintVariable {
+	return BlueprintVariable{
+		Name:        modVar.Name,
+		Description: modVar.Description,
+		Default:     modVar.Default,
+		Required:    modVar.Required,
+		VarType:     modVar.Type,
+	}
+}
+
+//build output
+func getBlueprintOutput(modOut *tfconfig.Output) BlueprintOutput {
+	return BlueprintOutput{
+		Name:        modOut.Name,
+		Description: modOut.Description,
+	}
+}
+
+// getBlueprintRequirements gets the variables and outputs associated
+// with the blueprint
+func getBlueprintRequirements(rolesConfigPath, servicesConfigPath string) BlueprintRequirements {
+	//parse blueprint roles
+	p := hclparse.NewParser()
+	rolesFile, _ := p.ParseHCLFile(rolesConfigPath)
+	r := parseBlueprintRoles(rolesFile)
+
+	//parse blueprint services
+	servicesFile, _ := p.ParseHCLFile(servicesConfigPath)
+	s := parseBlueprintServices(servicesFile)
+
+	return BlueprintRequirements{
+		Roles:    r,
+		Services: s,
+	}
+}
+
+// parseBlueprintRoles gets the roles required for the blueprint to be provisioned
+func parseBlueprintRoles(rolesFile *hcl.File) []BlueprintRoles {
+	var r []BlueprintRoles
+	iamContent, _, _ := rolesFile.Body.PartialContent(rootSchema)
+	for _, block := range iamContent.Blocks {
+		if block.Type == "locals" {
+			iamAttrs, _ := block.Body.JustAttributes()
+			for k, _ := range iamAttrs {
+				var iamRoles []string
```

Comment:
This is checking for all attributes defined under `locals`. For some blueprints we have more than one e.g. [GKE](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/test/setup/iam.tf#L17)

Reply:
Gotcha,  I am wondering if we should just align GKE to normal standards. We can keep this for now and think through the UX. My initial reasoning was that I was planning to expand the folder/org iam following something like https://github.com/terraform-google-modules/terraform-google-event-function/blob/78dc870d7e8861ef118dc43990590c9cfd78ee34/test/setup/iam.tf#L29-L39 which would require us to look at keys.

Reply:
Yeah, we'll probably need to do that because the way it is right now, the api to isn't able decipher the associated level i.e. org, folder or project. So I think have a key in the attribute itself would make things easier. Created b/249673472 to track this